src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,210 @@
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        Node<T> node = findNodeByIndex(index);
+        T value = node.value;
+        deleteNode(node);
+        return value;
     }
 
     @Override
     public boolean remove(T object) {
+        if (object == null) {
+            for (Node<T> node = first; node != null; node = node.next) {
+                if (node.value == null) {
+                    deleteNode(node);
+                    return true;
+                }
+            }
+        } else {
+            for (Node<T> node = first; node != null; node = node.next) {
+                if (node.value != null && node.value.equals(object)) {
+                    deleteNode(node);
+                    return true;
+                }
+            }
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    public T getFirst() {
+        if (first == null) {
+            throw new NoSuchElementException();
+        }
+        return first.value;
+    }
+
+    public T getLast() {
+        if (last == null) {
+            throw new NoSuchElementException();
+        }
+        return last.value;
+    }
+
+    private void deleteNode(Node<T> node) {
+        Node<T> prevNode = node.prev;
+        Node<T> nextNode = node.next;
+        if (prevNode == null) {
+            first = nextNode;
+        } else {
+            prevNode.next = nextNode;
+            node.next = null;
+        }
+        if (nextNode == null) {
+            last = prevNode;
+        } else {
+            nextNode.prev = prevNode;
+            node.prev = null;
+        }
+        --size;
+    }
+
+    private Node<T> findNodeByIndex(int index) {
+        if (index < (size / 2)) {
+            Node<T> node = first;
+            for (int i = 0; i < index; ++i) {
+                node = node.next;
+            }
+            return node;
+        } else {
+            Node<T> node = last;
+            for (int i = size - 1; i > index; --i) {
+                node = node.prev;
+            }
+            return node;
+        }
+    }
+
+    private void checkIndex(int index) {
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException("Index " + index + ", Size " + size);
+        }
+    }
+
+    private void checkIndexForAdd(int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("Index " + index + ", Size " + size);
+        }
+    }
+
+    public static class Node<T> {
+        private T value;
+        private Node prev;
+        private Node next;
+
+        public Node(Node prev, T value, Node next) {
+            this.prev = prev;
+            this.value = value;
+            this.next = next;
+        }
+
+        /*Message for Mentor
```

Comment:
you don't need accessor methods, I do not remember such rule

Reply:
please send a screenshot with a message that says about that rule violation

Reply:
> please send a screenshot with a message that says about that rule violation

![photo_2024-04-16_23-03-24](https://github.com/mate-academy/jv-linked-list/assets/72004177/1b5c0830-1a80-4370-893a-78c3540aa16d)